From 36ca464e2e9c230774ced9466482e79d67f7cd0a Mon Sep 17 00:00:00 2001
From: Dave Stevenson <dave.stevenson@raspberrypi.com>
Date: Mon, 4 Jul 2022 18:12:56 +0100
Subject: [PATCH] media: i2c: ov9281: Change exposure default value with mode
 choice

When the exposure range is updated due to changing VBLANK, adopt
the mode default value, not the current default.

Should the specific default value be outside of the min/max range
defined, then the whole update is rejected, which can leave an
invalid exposure range active.

Signed-off-by: Dave Stevenson <dave.stevenson@raspberrypi.com>
---
 drivers/media/i2c/ov9281.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/media/i2c/ov9281.c b/drivers/media/i2c/ov9281.c
index 2fb40e63c585..552b65847953 100644
--- a/drivers/media/i2c/ov9281.c
+++ b/drivers/media/i2c/ov9281.c
@@ -976,7 +976,7 @@ static int ov9281_set_ctrl(struct v4l2_ctrl *ctrl)
 		__v4l2_ctrl_modify_range(ov9281->exposure,
 					 ov9281->exposure->minimum, max,
 					 ov9281->exposure->step,
-					 ov9281->exposure->default_value);
+					 ov9281->cur_mode->vts_def);
 		break;
 	}
 
-- 
2.32.0

